### PR TITLE
Add sid claim to context

### PR DESCRIPTION
**Pull request by ThaminduR**
### Proposed changes in this pull request

- Add SId claim coming from the federated identity provider to authentication context

Related to https://github.com/wso2/product-is/issues/10714

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java [hunk 1]
```diff
@@ -477,6 +479,12 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                             errorMessage);
                 }
 
+                String idpName = context.getExternalIdP().getIdPName();
+                String sidClaim = (String) jsonObject.get(OIDCAuthenticatorConstants.SID);
+                if (StringUtils.isNotBlank(sidClaim) && StringUtils.isNotBlank(idpName)) {
+                    context.setProperty(FEDERATED_IDP_SESSION_ID + idpName, sidClaim);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:485]: Let's add a comment explaining why we are adding this here?

Comment (ThaminduR) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:487]: Added in the 84d75bb2a1e10818cb7c37f084787a25267cff11

Comment (mefarazath) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:487]: Also shouldn't this be set as a parameter not as a property?

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java [hunk 2]
```diff
@@ -477,6 +479,14 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                             errorMessage);
                 }
 
+                //Add SId claim into authentication context, to be stored in the UserSessionStore for
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:482]:
```suggestion
                // Add 'sid' claim into authentication context, to be stored in the UserSessionStore for
```

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java [hunk 3]
```diff
@@ -477,6 +479,14 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                             errorMessage);
                 }
 
+                String idpName = context.getExternalIdP().getIdPName();
+                String sidClaim = (String) jsonObject.get(OIDCAuthenticatorConstants.SID);
+                if (StringUtils.isNotBlank(sidClaim) && StringUtils.isNotBlank(idpName)) {
+                    //Add 'sid' claim into authentication context, to be stored in the UserSessionStore for
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:485]:
```suggestion
                    // Add 'sid' claim into authentication context, to be stored in the UserSessionStore for
```

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java [hunk 4]
```diff
@@ -477,6 +479,14 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                             errorMessage);
                 }
 
+                String idpName = context.getExternalIdP().getIdPName();
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:482]: In a federated scenario, is there any change for context.getExternalIdP() to return a null?

**Review by mefarazath — APPROVED** [review]

**Pull request merged**